NetworkingComparison/RxSwift/RxSwiftAPI.swift

```diff
@@ -0,0 +1,121 @@
+//
+//  RxSwiftAPI.swift
+//  NetworkingComparison
+//
+//  Created by Ian Luo on 11/18/19.
+//  Copyright © 2019 Lokae. All rights reserved.
+//
+
+import Foundation
+import RxSwift
+import RxCocoa
+
+class RxSwiftAPI {
+    static let shared = RxSwiftAPI()
+
+    enum RxSwiftError: Error {
+        case badUrl, invalidKey, cityNotFound, serverFailure
+    }
+
+    private let session = URLSession.shared
+
+    private enum SessionStyle {
+        case observableRequest, standardRequest, manualResponse
+    }
+
+    private var sessionStyle: SessionStyle = .observableRequest
+
+    private init() { }
+
+    func getForecasts() -> Observable<[CodableForecast]> {
+        var data: Observable<Data>
+        switch sessionStyle {
+        case .observableRequest: data = fetchWithStandardRequest()
+        case .standardRequest: data = fetchWithObservableRequest()
+        case .manualResponse: data = fetchWithManualResponse()
+        }
+        return data.map {
+            let decoder = JSONDecoder()
+            decoder.dateDecodingStrategy = .secondsSince1970
+            return try decoder.decode(CodableForecastWrapper.self, from: $0).forecasts
+        }
+    }
+
+    private func fetchWithStandardRequest() -> Observable<Data> {
+        print("********** RxSwift requesting with standard URLRequest **********")
```

Comment:
It might be worth wrapping all of this in a `return Observable.deffered {` so that the configuration logic is only run once the returned `Observable` is subscribed to. I don't think it would have any effect in the current implementation but I would consider that a best practice (especially with networking code like this).

One case where this could be relevant is if you start introducing state to how your parameters are configured for the request like an Auth token that's updated elsewhere, you may want to ensure that anytime this observable is resubscribe to it uses the most up to date token.

Another is where you create this observable, but it's never actually subscribed to (like a ViewModel is instantiated that has a pointer to this observable which is subscribed to when the ViewModel becomes active, but it never becomes active). You likely wouldn't want to run this logic here, but instead only the it's subscribed to.